src/domain/community/profile/ContributionDetails/ContributionDetailsCard.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, Dialog } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import { Close } from '@mui/icons-material';
+import JourneyCard, { JourneyCardProps } from '../../../challenge/common/JourneyCard/JourneyCard';
+import { BlockTitle } from '../../../../core/ui/typography';
+import webkitLineClamp from '../../../../core/ui/utils/webkitLineClamp';
+import CardActions from '../../../../core/ui/card/CardActions';
+import JourneyCardTagline from '../../../challenge/common/JourneyCard/JourneyCardTagline';
+import { DialogActions, DialogContent, DialogTitle } from '../../../../common/components/core/dialog';
+import { JourneyTypeName } from '../../../challenge/JourneyTypeName';
+import { OpportunityIcon } from '../../../challenge/opportunity/icon/OpportunityIcon';
+import { ChallengeIcon } from '../../../challenge/challenge/icon/ChallengeIcon';
+import { HubIcon } from '../../../challenge/hub/icon/HubIcon';
+
+interface ContributionDetailsCardProps extends Omit<JourneyCardProps, 'iconComponent' | 'header'> {
+  journeyTypeName: JourneyTypeName;
+  displayName: string;
+  enableLeave?: boolean;
+  leavingCommunityDialogOpen?: boolean;
+  leavingCommunity?: boolean;
+  onLeaveCommunityDialogOpen?: (isOpen: boolean) => void;
+  handleLeaveCommunity?: () => void;
+  loading?: boolean;
+}
+
+const getIconComponent = (journeyTypeName: JourneyTypeName) => {
+  switch (journeyTypeName) {
+    case 'hub':
+      return HubIcon;
+    case 'challenge':
+      return ChallengeIcon;
+    case 'opportunity':
+      return OpportunityIcon;
+  }
+};
```

Comment:
I creates JourneyType enum and journeyIcon function that returns the appropriate icon. (followed the pattern from TopCallouts). Maybe consider checking that out here https://github.com/alkem-io/client-web/pull/3220/files#diff-0f0bc702443c7537102d0ec931621846e2baf8aa659e878cf9dd200813fb3719